ipaserver/install/ipa_idrange_fix.py

```diff
@@ -0,0 +1,1073 @@
+                f"{self.dn})"
+            )
+        return (
+            f"group(groupname='{self.name}', "
+            f"gid={self.number}, "
+            f"{self.dn})"
+        )
+
+    def __eq__(self, other):
+        return self.number == other.number
+
+
+class IPAIDRangeFix(AdminTool):
+    """Tool to analyze and fix IPA ID ranges"""
+
+    command_name = "ipa-idrange-fix"
+    log_file_name = "/var/log/ipa-idrange-fix.log"
+    usage = "%prog"
+    description = "Analyze and fix IPA ID ranges"
+
+    @classmethod
+    def add_options(cls, parser, debug_option=False):
+        super(IPAIDRangeFix, cls).add_options(parser)
+        parser.add_option(
+            "--ridoffset",
+            dest="ridoffset",
+            type=int,
+            default=100000,
+            metavar=100000,
+            help="Offset for a next base RID from previous RID range. \
+Needed for future range size expansions. Has to be > 0",
+        )
+        parser.add_option(
+            "--rangegap",
+            dest="rangegap",
+            type=int,
+            default=200000,
+            metavar=200000,
+            help="Threshold for a gap between out-of-range IDs to be \
+considered a different range. Has to be > 0",
+        )
+        parser.add_option(
+            "--minrange",
+            dest="minrange",
+            type=int,
+            default=10,
+            metavar=10,
+            help="Minimal considered range size for out-of-range IDs.\
+All ranges with amount of IDs lower than this number will be discarded and \
+IDs will be listed to be moved manually. Has to be > 1",
+        )
+        parser.add_option(
+            "--allowunder1000",
+            dest="allowunder1000",
+            action="store_true",
+            default=False,
+            help="Allow idranges to start below 1000. Be careful to not \
+overlap IPA users/groups with existing system-local ones!",
+        )
+        parser.add_option(
+            "--norounding",
+            dest="norounding",
+            action="store_true",
+            default=False,
+            help="Disable IDrange rounding attempt in order to get ranges \
+exactly covering just IDs provided",
+        )
+        parser.add_option(
+            "--unattended",
+            dest="unattended",
+            action="store_true",
+            default=False,
+            help="Automatically fix all range issues found without asking \
+for confirmation",
+        )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.realm: str = None
+        self.suffix: DN = None
+        self.propositions_rid: list[IDRange] = []
+        self.propositions_new: list[IDRange] = []
+        self.outliers: list[IDentity] = []
+        self.under1000: list[IDentity] = []
+        self.id_ranges: list[IDRange] = []
+
+    def validate_options(self, needs_root=True):
+        # super(IPAIDRangeFix, self).validate_options(needs_root)
```

Comment:
Did you intend to retain the commented-out super() call?

Reply:
no, removed.